events/equeue/equeue_mbed.cpp

```diff
@@ -131,7 +131,7 @@ static void equeue_sema_timeout(equeue_sema_t *s) {
 bool equeue_sema_wait(equeue_sema_t *s, int ms) {
     int signal = 0;
     Timeout timeout;
-    if (ms > 0) {
+    if (ms >= 0) {
```

Comment:
That is the current behaviour of this pr. When ms is `0`, the ticker attach should interrupt immediately and set the flag to `1`, causing the following while loop ([here](https://github.com/ARMmbed/mbed-os/pull/3991/files/d836520f21e99b10e1d1d34fab51225dd4a902f3#diff-fabf0dde2c29a3a1fe9a3f4f83632addR139)) to exit immediately.

Do you think the zero case behaviour should be explicit?
``` diff
-    if (ms > 0) {
+    if (ms == 0) {
+        return false;
+    } else if (ms > 0) {
         timeout.attach_us(callback(equeue_sema_timeout, s), ms*1000);
     }
```